lib/ansible/module_utils/facts/virtual/linux.py

```diff
@@ -139,19 +139,32 @@ def get_virtual_facts(self):
             virtual_facts['virtualization_role'] = 'guest'
 
         product_name = get_file_content('/sys/devices/virtual/dmi/id/product_name')
+        sys_vendor = get_file_content('/sys/devices/virtual/dmi/id/sys_vendor')
 
         if product_name in ('KVM', 'KVM Server', 'Bochs', 'AHV'):
             guest_tech.add('kvm')
             if not found_virt:
                 virtual_facts['virtualization_type'] = 'kvm'
                 found_virt = True
 
-        if product_name == 'RHEV Hypervisor':
-            guest_tech.add('RHEV')
+        if sys_vendor == 'oVirt':
+            guest_tech.add('oVirt')
             if not found_virt:
-                virtual_facts['virtualization_type'] = 'RHEV'
+                virtual_facts['virtualization_type'] = 'oVirt'
                 found_virt = True
 
+        if sys_vendor == 'Red Hat':
+            if product_name == 'RHEV Hypervisor':
+                guest_tech.add('RHEV')
+                if not found_virt:
+                    virtual_facts['virtualization_type'] = 'RHEV'
+                    found_virt = True
+            elif product_name == 'RHEL':
+                guest_tech.add('RHV')
+                if not found_virt:
+                    virtual_facts['virtualization_type'] = 'RHV'
+                    found_virt = True
```

Comment:
we can also consider checking if ovirt-guest-agent runs (as we check vdsmd on the host) to cover VMs that started on RHV 4.3 and keep running on 4.3 cluster in RHV 4.4

Reply:
I would avoid agent checks...if we can figuer out everything based on static immutable smbios data then let's do it that way. it's really not that complicated, the differences between versions are finite (sometimes it's also just a bug on ovirt side - like https://bugzilla.redhat.com/show_bug.cgi?id=1931786)

Reply:
though, right, the only case where smbios data is not good enough is 4.3 rhel host where it looks exactly the same as on plain virt (martin's table for rhv 4.4 cluster level 4.3)
I guess I would rather leave it be and identify as KVM and consider it a "bug" fixed in later CL

Reply:
I don't mind much but if it's not too complicated then I would check for ovirt-guest-agent in a lazy way, meaning that we first check for product family (assuming we'll set it to all VMs that are started by RHVM 4.4) and if it's not set, check if ovirt-guest-agent is installed to cover everything. if it's complicated to do then yeah, I'd skip the last part

Reply:
Can't do a change in RHV Manager 4.4, which would set proper product family to RHV regardess of cluster version? Meaning that even 4.2 VMs would have product family set correctly. As RHV Manager 4.2/4.3 are already out of support, I'd not care that 4.2 VM spawned by 4.2 RHVM has different product family than spawned by 4.4 RHVM

Reply:
it's not out of support, it's out of development, but it's still perfectly usable. I'd really want to avoid change of guest ABI in existing versions. That virtio socket check is trivial, just a file existence...

Reply:
What file would we need to check?

Reply:
/dev/virtio-ports/ovirt-guest-agent.0 would be for OGA but we dropped it in el8/ubuntu18 in RHV 4.3 already I think.
/org.qemu.guest_agent.0 we add always...but @nyoxi are we the only ones using it with this name?
there's also /com.redhat.spice.0 but only when SPICE is defined, that is still a default.

Reply:
Martin, to your question, there are really two approaches here:
1. the conservative approach that says VMs on 4.3 cluster-level with RHVM 4.4 should be exactly as if they were initiated by RHVM 4.3. this means those VMs should still be set with product-family = RHEL.
2. the progressive approach that considers exceptions to this rule when they make sense. for example, we removed the OGA channel also for RHEL 8 guests also on 4.3 cluster-level with RHVM 4.4. in theory it broke the ABI but no one really cares..

I tend to think we can go with the progressive approach here (posted a patch for that) - and then checking product-family covers all VMs that are started by RHVM 4.4 (regardless of their cluster level) and the check for OGA channel (which was dropped in 4.4.2 btw - https://bugzilla.redhat.com/show_bug.cgi?id=1845591) covers most of the VMs that were started by RHVM < 4.4.

but we didn't reach an agreement on the ability to take the progressive approach here. the only winning argument I can see for this is showing that changing the product family has no real impact (like dropping OGA channel on RHEL 8 guest in 4.3 cluster level) but unfortunately I can't prove that.
so I guess we can fall back to the conservative approach which leaves us with the identification problem for VMs on 4.3 cluster-level and keep exploring options like Michal suggested in his last comment.

Reply:
> /dev/virtio-ports/ovirt-guest-agent.0 would be for OGA but we dropped it in el8/ubuntu18 in RHV 4.3 already I think.

From what I have found it seems that the channel name was:

* before CL 4.2 -- named `com.redhat.rhevm.vdsm`
* in cluster level 4.2 -- renamed to `ovirt-guest-agent.0`
* oVirt 4.4.2 -- removed completely from EL8 and CoreOS VMs (for all cluster levels)

> /org.qemu.guest_agent.0 we add always...but @nyoxi are we the only ones using it with this name?
there's also /com.redhat.spice.0 but only when SPICE is defined, that is still a default.

The names for spice and qemu-ga channels are de-facto standards. It is not specific to oVirt.

Reply:
yeah, so with that i would conclude that we have no good way on 4.2/4.3 and leave it broken on RHEL (and it does work on 4.2/4.3 RHVH at least)
And finally merge this, it's worse that we keep it broken entirely on a current version for 3 months now:)